### PR TITLE
feat(toolbar): add vertical variant

### LINKED ISSUE #806: Toolbar — orientation
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-orientation_attribute

### PR DESCRIPTION
**Pull request by SimonFinney**
#### Affected issues

- Contributes to #67 
- Resolves #806 

#### Proposed changes

- Add toolbar vertical variant logic, stories, styles, and test
- Update default toolbar story to better reflect [guidance](https://pages.github.ibm.com/cdai-design/pal/patterns/toolbars/#responsiveness)
- Refactor common toolbar tests to better accommodate individual component tests

### CODE REVIEW

File: packages/cloud-cognitive/src/components/Toolbar/_toolbar.scss [hunk 1]
```diff
@@ -5,38 +5,55 @@
...
 @include exports($name: cloud-cognitive--toolbar) {
   $block-class: #{$pkg-prefix}--toolbar;
+
   $border: carbon--rem(
       $px: 1px,
     )
     solid $ui-03;
 
+  $dimensions: carbon--rem(
+    $px: 40px,
```
Comment (matthewgallo) [inline packages/cloud-cognitive/src/components/Toolbar/_toolbar.scss:21]: I believe you could just use `$spacing-08` here

**Comment by matthewgallo** [comment]
https://user-images.githubusercontent.com/10215203/130460308-af865b1d-ed6c-4840-96d0-a49abc9c2549.mov
For the vertical toolbar, it might be easier to use if the tooltip has the option to be placed from the left or right. You cannot easily hover over the item above the current toolbar item because the tooltip is on top of it.

File: packages/cloud-cognitive/src/components/Toolbar/_toolbar.scss (hunk 1 above)
Comment (SimonFinney) [reply packages/cloud-cognitive/src/components/Toolbar/_toolbar.scss:21]: Thanks — would we typically use spacing tokens for properties other than [margin and padding](https://carbondesignsystem.com/guidelines/spacing/overview/#applying-spacing)? As I understand it, Carbon mostly favors [`rem` for height and width dimensions](https://github.com/search?q=%27width%3A%27+path%3Apackages%2Fstyles+repo%3Acarbon-design-system%2Fcarbon&type=Code). What's the current thinking here?

Comment (matthewgallo) [reply packages/cloud-cognitive/src/components/Toolbar/_toolbar.scss:21]: You’re totally right! Nevermind then, what you had is good

**Comment by SimonFinney** [comment]
> https://user-images.githubusercontent.com/10215203/130460308-af865b1d-ed6c-4840-96d0-a49abc9c2549.mov
> For the vertical toolbar, it might be easier to use if the tooltip has the option to be placed from the left or right. You cannot easily hover over the item above the current toolbar item because the tooltip is on top of it.

@matthewgallo a quick update on this — the design recommendation is that the default tooltip position should be `top` for both orientations. My gut here is to create an issue and reduced test case in Carbon to investigate further, as it feels like a defect that may not affect just us.

**Review by matthewgallo — APPROVED** [review]

**Pull request merged**